ec2_metadata.py

```diff
@@ -61,6 +61,28 @@ def mac(self):
         return requests.get(METADATA_URL + 'mac').text
 
     @cached_property
+    def network(self):
```

Comment:
so then I/we have to create a network_interfaces object? Any tips on how to do that in a sane way?

Name- your choice. It's more like "network metadata by interface", I took "network" because that's the first part of the URI.

I'll paste the output in the PR soon.